.github/workflows/deploy-lambda-code.yaml

```diff
@@ -40,6 +54,52 @@ on:
         type: string
 
 jobs:
+  derive-service-names:
+    name: "Derive Service Names"
+    runs-on: ubuntu-latest
+    outputs:
+        services: ${{ steps.derive-service-names.outputs.services }}
+    steps:
+      - name: "Checkout code"
+        uses: NHSDigital/uec-dos-management/.github/actions/checkout-repository@v0.32
+        with:
+          repository: ${{ inputs.domain }}
+          tag: ${{ inputs.tag }}
+
+      - name: Derive Service Names
+        id: derive-service-names
+        shell: bash
+        run: |
+          if [ ${{ inputs.services }} == 'all' ]; then
+            cd ${{ inputs.directory }}
+            SERVICES=$(ls | jq -R -s -c 'split("\n")[:-1]')
+            echo "services=${SERVICES}" >> $GITHUB_OUTPUT
+          else
+            echo "services=${{ inputs.services }}" >> $GITHUB_OUTPUT
+          fi
+
+  derive-commit-hash:
+    name: Derive Commit Hash For Lambda Code
+    runs-on: ubuntu-latest
+    outputs:
+        commit_hash: ${{ steps.derive-commit-hash.outputs.commit_hash }}
+    steps:
+      - name: "Checkout code"
+        uses: NHSDigital/uec-dos-management/.github/actions/checkout-repository@v0.32
+        with:
+          repository: ${{ inputs.domain }}
+
+      - name: Derive Commit Hash For Lambda Code
```

Comment:
Maybe one for later but can we refactor this as the same code appears in the react deploy section and it feels like we could maybe create a reusable callable action/flow to do this

Reply:
Ticket to be created

Reply:
DR-1028 to cover refactoring